toqito/matrix_props/is_pseudohermitian.py

```diff
@@ -0,0 +1,34 @@
+def is_pseudohermitian(mat: np.ndarray, signature_tup: tuple):
+    """
+    Checks if a matrix is pseudo-hermitian with respect to a given signature.
+
+    Args:
+        mat (np.ndarray): The matrix to check.
+        signature_tup (tuple): A tuple (p, q) where p is the number of +1's and q is the number of -1's in the metric.
+
+    Returns:
+        bool: True if the matrix is pseudo-hermitian, False otherwise.
+    """
+    # Check if the matrix is square
+    if mat.shape[0] != mat.shape[1]:
```

Comment:
We have a function for this that you can import from `toqito` (`from toqito.matrix_props import is_square`)